src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -107,16 +110,24 @@ class BlockingProcess(
     }
 
     try {
-      val stdin =
-        context.actorOf(OutputStreamSubscriber.props(process.getOutputStream), "stdin")
-
-      val stdout =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getInputStream, stdioTimeout), "stdout"))
-
-      val stderr =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getErrorStream, stdioTimeout), "stderr"))
-
-      context.parent ! Started(ActorSubscriber(stdin), ActorPublisher(stdout), ActorPublisher(stderr))
+      val stdin = Sink.outputStream(process.getOutputStream)
+        .mapMaterializedValue { futureResult =>
+          futureResult.onComplete(_ => context.stop(self))
```

Comment:
That does actually sound better - I'll have 3 messages `StdinTerminated`, `StdoutTerminated`, and `StderrTerminated` sent at the end of stdin Sink, stdout Source, and stderr Source.

Once either one of these message is received, then I'll handle it the same way as `Destroy` and `Terminated`